cfme/cloud/stack.py

```diff
@@ -41,197 +261,197 @@ def __init__(self, name, provider, quad_name=None, appliance=None):
-
         # Navigate to the starting destination
         if from_dest in navigator.list_destinations(self):
-            navigate_to(self, from_dest)
+            view = navigate_to(self, from_dest)
         else:
             msg = 'cfme.cloud.stack does not have destination {}'.format(from_dest)
             raise DestinationNotFound(msg)
 
         # Delete using the method appropriate for the starting destination
         if from_dest == 'All':
-            sel.check(Quadicon(self.name, self.quad_name).checkbox())
-            cfg_btn("Remove Orchestration Stacks", invokes_alert=True)
+            view.toolbar.view_selector.select('List View')
+            try:
+                row = view.paginator.find_row_on_pages(view.entities.table, name=self.name)
+                row[0].check()
+                view.toolbar.configuration.item_select('Remove Orchestration Stacks',
+                                                       handle_alert=True)
+            except NoSuchElementException:
+                raise StackNotFound('Stack {} not found'.format(self.name))
+            view.entities.flash.assert_no_error()
+            view.entities.flash.assert_success_message(
+                'Delete initiated for 1 Orchestration Stacks from the CFME Database')
         elif from_dest == 'Details':
-            cfg_btn("Remove this Orchestration Stack", invokes_alert=True)
+            view.toolbar.configuration.item_select('Remove this Orchestration Stack',
+                                                   handle_alert=True)
+            view.entities.flash.assert_no_error()
+            view.entities.flash.assert_success_message('Delete initiated for 1 Orchestration Stack')
 
-        sel.handle_alert()
-        # The delete initiated message may get missed if the delete is fast
-        try:
-            flash.assert_message_contain("Delete initiated for 1 Orchestration Stacks")
-        except FlashMessageException as ex:
-            if 'No flash message contains' in ex.message:
-                flash.assert_message_contain("The selected Orchestration Stacks was deleted")
+        def refresh():
+            """Refresh the view"""
+            if self.provider:
+                self.provider.refresh_provider_relationships()
+            view.browser.selenium.refresh()
+            view.flush_widget_cache()
 
-        self.wait_for_delete()
+        wait_for(lambda: not self.exists, fail_condition=False, fail_func=refresh, num_sec=15 * 60,
+                 delay=30, message='Wait for stack to be deleted')
 
     def edit_tags(self, tag, value):
-        navigate_to(self, 'EditTags')
-        fill(edit_tags_form, {'select_tag': tag,
-                              'select_value': value},
-             action=form_buttons.save)
-        flash.assert_success_message('Tag edits were successfully saved')
+        """Edit the tags of a particular stack
+        :param tag: The tag to set
+        :param value: The value of the tag
+        """
```

Comment:
**REQUIRED**
Inconsistency is here. Below you used `Args` section for arguments and here `:param`.